week3/api-services/security_integration.py

```
@@ -0,0 +1,359 @@
+            if response.status_code == 200:
+                result = response.json()
+                # Cache successful validation
+                self.auth_cache[cache_key] = (result, time.time())
+                return result
+            else:
+                # Cache negative result briefly
+                self.auth_cache[cache_key] = (None, time.time())
+                return None
+                
+        except requests.RequestException as e:
+            logger.error(f"Security service unreachable: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)